perf/perf_hv_gpci.py

```diff
@@ -82,17 +95,60 @@ def run_cmd(self, cmd):
 
     def test_gpci_events(self):
         perf_stat = "perf stat"
-        perf_flags = '-C 1 -v -e'
+        perf_flags = '-v -e'
 
-        for line in self.list_of_hv_gpci_events:
-            evt = "hv_gpci/%s,hw_chip_id=12/" % line
-            cmd = "%s %s %s sleep 1" % (perf_stat, perf_flags, evt)
+        for line in self.list_phys:
+            line = "%s,%s/" % (line.split(',')[0], line.split(',')[1].replace(
+                'phys_processor_idx=?', 'phys_processor_idx=1'))
+            cmd = "%s %s %s sleep 1" % (perf_stat, perf_flags, line)
             self.run_cmd(cmd)
-            cmd = "%s --per-core -a -e %s sleep 1" % (perf_stat, evt)
+            cmd = "%s --per-core -a -e %s sleep 1" % (perf_stat, line)
             self.run_cmd(cmd)
-            cmd = "%s --per-socket -a -e %s sleep 1" % (perf_stat, evt)
+            cmd = "%s --per-socket -a -e %s sleep 1" % (perf_stat, line)
             self.run_cmd(cmd)
+        self.error_check()
+
+        for line in self.list_sibling:
+            line = "%s,%s/" % (line.split(',')[0], line.split(',')[1].replace(
+                'sibling_part_id=?', 'sibling_part_id=2'))
+            cmd = "%s %s %s sleep 1" % (perf_stat, perf_flags, line)
+            self.run_cmd(cmd)
+            cmd = "%s --per-core -a -e %s sleep 1" % (perf_stat, line)
+            self.run_cmd(cmd)
+            cmd = "%s --per-socket -a -e %s sleep 1" % (perf_stat, line)
+            self.run_cmd(cmd)
+        self.error_check()
 
+        for line in self.list_partition:
+            line = "%s,%s/" % (line.split(',')[0], line.split(',')[1].replace(
+                'partition_id=?', 'partition_id=1'))
+            cmd = "%s %s %s sleep 1" % (perf_stat, perf_flags, line)
+            self.run_cmd(cmd)
+            cmd = "%s --per-core -a -e %s sleep 1" % (perf_stat, line)
+            self.run_cmd(cmd)
+            cmd = "%s --per-socket -a -e %s sleep 1" % (perf_stat, line)
+            self.run_cmd(cmd)
+        self.error_check()
+
+        for line in self.list_hw:
+            line = "%s,%s/" % (line.split(',')[0], line.split(',')[1].replace(
+                'hw_chip_id=?', 'hw_chip_id=12'))
+            cmd = "%s %s %s sleep 1" % (perf_stat, perf_flags, line)
+            self.run_cmd(cmd)
+            cmd = "%s --per-core -a -e %s sleep 1" % (perf_stat, line)
+            self.run_cmd(cmd)
+            cmd = "%s --per-socket -a -e %s sleep 1" % (perf_stat, line)
+            self.run_cmd(cmd)
+        self.error_check()
+
+        for line in self.list_noid:
+            line = "%s/" % line
+            cmd = "%s %s %s sleep 1" % (perf_stat, perf_flags, line)
+            self.run_cmd(cmd)
+            cmd = "%s --per-core -a -e %s sleep 1" % (perf_stat, line)
+            self.run_cmd(cmd)
+            cmd = "%s --per-socket -a -e %s sleep 1" % (perf_stat, line)
+            self.run_cmd(cmd)
```

Comment:
@disgoel I feel using nested for_loop, we can optimize the code. most of them looks redundant.
let me know if you need help on it